libclang was introduced in
https://github.com/tensorflow/tensorflow/commit/c211472000ff57bac7fcec9b0465cf73b37bf135
> This is in preparation to open-source TF's TFRT backend.
> TFRT generates code using libclang python bindings as part of the build.
Hence it is not currently used and as it is not (easily) available for all architectures
simply remove it.

--- tensorflow/tools/pip_package/setup.py.tpl.orig	2026-06-29 00:00:00.000000000 +0200
+++ tensorflow/tools/pip_package/setup.py.tpl	2026-06-29 00:00:00.000000000 +0200
@@ -97,7 +97,6 @@
     'flatbuffers >= 25.9.23',
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
-    'libclang >= 13.0.0',
     'opt_einsum >= 2.3.2',
     'packaging',
     'protobuf >= 6.31.1, < 8.0.0',
